### PR TITLE
Add vol scan

### PR DESCRIPTION
**Pull request by ernstdevreede**
This adds functionality for visualising KNMI radar HDF5 volume scan data.
The code generates parameters for the following parameters: "KDP", "PhiDP", "RhoHV", "V", "W", "Z" with a dimension named `scan_elevation` with values 0.3, 0.8, 1.2, 2, 2.8, 4.5, 6, 8, 10, 12, 15, 20, 25 coming from the scans numbered {7, 6, 14, 5, 13, 4, 12, 3, 11, 2, 10, 9, 8} in the HDF5 file.
The generated parameters can be used as NetCDF variables in the layer configuration.

### CODE REVIEW

File: CCDFDataModel/CCDFHDF5IO.h [hunk 1]
```diff
@@ -118,6 +123,171 @@ class CDFHDF5Reader : public CDFReader {
   bool b_KNMIHDF5UseEndTime;
 
 public:
+  class CustomVolScanReader : public CDF::Variable::CustomReader {
+  public:
+    ~CustomVolScanReader() {}
+
+    int getCalibrationParameters(CT::string formula, float &factor, float &offset) {
```
Comment (maartenplieger) [inline CCDFDataModel/CCDFHDF5IO.h]: This should go to the CPP file if no templates are used

File: CCDFDataModel/CCDFHDF5IO.h [hunk 2]
```diff
@@ -118,6 +123,171 @@ class CDFHDF5Reader : public CDFReader {
   bool b_KNMIHDF5UseEndTime;
 
 public:
+  class CustomVolScanReader : public CDF::Variable::CustomReader {
```
Comment (maartenplieger) [inline CCDFDataModel/CCDFHDF5IO.h:126]: This should go to its own file

File: adagucserverEC/CConvertKNMIH5VolScan.h [hunk 3]
```diff
@@ -0,0 +1,42 @@
+/******************************************************************************
+ *
+ * Project:  ADAGUC Server
+ * Purpose:  ADAGUC OGC Server
+ * Author:   Maarten Plieger, plieger "at" knmi.nl
```
Comment (maartenplieger) [inline adagucserverEC/CConvertKNMIH5VolScan.h]: Should be Ernst de Vreede

File: adagucserverEC/CConvertKNMIH5VolScan.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ * Date:     2013-06-01
```
Comment (maartenplieger) [inline adagucserverEC/CConvertKNMIH5VolScan.h]: New date would be nice too

**Review by maartenplieger — COMMENTED** [review]
The testfile is 40Mb, is there any change to make it smaller?

File: tests/AdagucTests/TestWMSVolScan.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+from lxml import etree, objectify
+import re
+from adaguc.AdagucTestTools import AdagucTestTools
+from lxml import etree, objectify
+import datetime
+
+ADAGUC_PATH = os.environ['ADAGUC_PATH']
+
+
+class TestWMSVolScan(unittest.TestCase):
```
Comment (maartenplieger) [inline tests/AdagucTests/TestWMSVolScan.py:17]: Can you also add a testcase for getlegendgraphic? The autowms getlegendgraphic on volume scans crashes

File: adagucserverEC/CConvertKNMIH5VolScan.cpp [hunk 6]
```diff
@@ -0,0 +1,453 @@
...
+ * limitations under the License.
+ *
+ ******************************************************************************/
+
+#include "CConvertKNMIH5VolScan.h"
+#include "CImageWarper.h"
+#include "COGCDims.h"
+#include "CCDFHDF5IO.h"
+
+#define CCONVERTKNMIH5VOLSCAN_DEBUG
```
Comment (maartenplieger) [inline adagucserverEC/CConvertKNMIH5VolScan.cpp]: This could be commented out for production

File: CCDFDataModel/CCDFHDF5IO.h (hunk 2 above)
Comment (ernstdevreede) [reply CCDFDataModel/CCDFHDF5IO.h:126]: Done

File: CCDFDataModel/CCDFHDF5IO.h (hunk 1 above)
Comment (ernstdevreede) [reply CCDFDataModel/CCDFHDF5IO.h]: done

File: adagucserverEC/CConvertKNMIH5VolScan.cpp (hunk 6 above)
Comment (ernstdevreede) [reply adagucserverEC/CConvertKNMIH5VolScan.cpp]: done

**Pull request merged**